src/Datadog.Trace/Tracer.cs

```diff
@@ -348,6 +349,40 @@ private static IStatsd CreateDogStatsdClient(TracerSettings settings, string ser
             return new Statsd(statsdUdp, new RandomGenerator(), new StopWatchFactory(), prefix: string.Empty, constantTags);
         }
 
+        private static void StartTraceAgentWhenNecessary()
+        {
+            var azureAppServicesValue = Environment.GetEnvironmentVariable("DD_AZURE_APP_SERVICES");
```

Comment:
I'm not sure under what conditions a security exception might happen in an Azure App Service, but `GetEnvironmentVariable` could potentially throw a `SecurityException` (see https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=netframework-4.8) so I suggest a try/catch in here.